### PR TITLE
Rename "Realm name" field label to "Realm ID" on create realm page

### LINKED ISSUE #23496: Rename "Realm name" field to "Realm ID" field in realm creation screen
### Description

Issue based on [this discussion post](https://github.com/keycloak/keycloak/issues/17735#issuecomment-1697830183).

When creating a new Realm, the field "Realm name" is a misnomer because it in fact refers to "Realm ID":

image

image

### PR DESCRIPTION
**Pull request by Anil-Shanker**
Closes #23496

### CODE REVIEW

File: js/apps/admin-ui/src/realm/add/NewRealmForm.tsx [hunk 1]
```diff
@@ -76,21 +76,21 @@ export default function NewRealmForm() {
...
               isRequired
-              id="kc-realm-name"
+              id="kc-realm-id"
               validated={errors.realm ? "error" : "default"}
               {...register("realm", {
                 required: { value: true, message: t("required") },
                 pattern: {
                   value: /^[a-zA-Z0-9-_]+$/,
-                  message: t("invalidRealmName"),
+                  message: t("invalidRealmId"),
```
Comment (jonkoops) [inline js/apps/admin-ui/src/realm/add/NewRealmForm.tsx:93]: Looking at the translations this field doesn't exist, you'll probably have to rename it in the JSON files like you did with the realm name.

**Review by jonkoops — CHANGES_REQUESTED** [review]

File: js/apps/admin-ui/src/realm-settings/GeneralTab.tsx [hunk 2]
```diff
@@ -94,7 +94,7 @@ export const RealmSettingsGeneralTab = ({
               required: { value: true, message: t("required") },
               pattern: {
                 value: /^[a-zA-Z0-9-_]+$/,
-                message: t("realm:invalidRealmName"),
+                message: t("realm:invalidRealmId"),
```
Comment (Anil-Shanker) [inline js/apps/admin-ui/src/realm-settings/GeneralTab.tsx:97]: Does passing "realm:invalidRealmId" refer to something other than "invalidRealmId"?

**Comment by Anil-Shanker** [comment]
I mistakenly had to add an extra commit - would it possible to squash the extra commits from inside GitHub?

**Review by jonkoops — CHANGES_REQUESTED** [review]

**Comment by jonkoops** [comment]
@Anil-Shanker are you still interested in landing this? Could you processes the review comments?

**Comment by ssilvert** [comment]
@Anil-Shanker I'm going to close this now.  We can reopen when you are ready to work on it again.

**Pull request closed without merging**